server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -604,32 +608,69 @@ public ClusterState addIndexTemplateV2(
             return currentState;
         }
 
-        validate(name, finalIndexTemplate);
-        validateDataStreamsStillReferenced(currentState, name, finalIndexTemplate);
+        validateIndexTemplateV2(name, finalIndexTemplate, currentState);
+        logger.info(
+            "{} index template [{}] for index patterns {}",
+            existing == null ? "adding" : "updating",
+            name,
+            template.indexPatterns()
+        );
+        return ClusterState.builder(currentState).metadata(Metadata.builder(currentState.metadata()).put(name, finalIndexTemplate)).build();
+    }
+
+    private void validateIndexTemplateV2(String name, ComposableIndexTemplate indexTemplate, ClusterState currentState) {
+        // Workaround for the fact that start_time and end_time are injected by the MetadataCreateDataStreamService upon creation,
+        // but when validating templates that create data streams the MetadataCreateDataStreamService isn't used.
+        ComposableIndexTemplate templateToValidate = indexTemplate;
+        if (indexTemplate.getDataStreamTemplate() != null) {
+            var result = resolveIndexModeAndLookAheadTimeSetting(indexTemplate, currentState.metadata());
+            var indexMode = result.v1();
+            if (indexMode == IndexMode.TIME_SERIES) {
+                Template finalTemplate = indexTemplate.template();
+                Settings.Builder indexSettings = Settings.builder().put(finalTemplate.settings());
+                MetadataCreateDataStreamService.prepareFirstBackingIndexForTSDB(indexSettings, System.currentTimeMillis(), result.v2());
```

Comment:
Should these be switched? I'm thinking of the case when someone includes the TSDB time settings in their template itself (which, I realize, is not recommended, but in the interest of correctness). Maybe something like:

```java
Settings.Builder indexSettings = Settings.builder();
MetadataCreateDataStreamService.prepareFirstBackingIndexForTSDB(indexSettings, System.currentTimeMillis(), result.v2());
indexSettings.put(finalTemplate.settings());

... create the new template and validate ...
```

Reply:
I could be wrong on the order of index settings application in `MetadataCreateDataStreamService`, I'm not sure whether the "prepareFirstBackingIndexForTSDB" settings take precedence or not. Ignore me if I'm mistaken!

Reply:
Good point, I think the user provided start_time and end_time settings are now overwritten.
I can implement your suggestion and first apply the settings from `prepareFirstBackingIndexForTSDB(...)` method and then the `finalTemplate.settings()`.

As a side thought, I do wonder whether this will ever be beneficial to the user, because specifying absolute start_time and end_time values will eventually result in a weird result. For example not being able to index current metrics, because start and end time are too old. Maybe we should fail with an error here if custom start and end times are being provided (only if index.mode=time_series and data stream snippet is provided)?

Reply:
Eventually, we will need to support reindexing historical data. In this case some process or a user will specify start_time and end_time from the past.

Reply:
I see, then it makes sense to allow user defines start_time and end_time settings.